### DISCUSSION TITLE
Email templates on Demand

### DISCUSSION THREAD

**Discussion by imknight** [directus/directus#15924] [Archive]
I have a collection with email templates, and I would like to call the email templates in the flow, but it seems like those bracket variable {{ }} is not parsing, did I miss out anything or do I need to do anything so that when the flow sends out email will parse the variable ?

**Comment by azrikahar** [comment]
I believe in this particular scenario, using something like `{{read_email_template.body}}` to use the email template in a collection does not work due to the bracket/mustache variable is already technically parsed (it is actually parsed to get the read data operation email template, but not double parsed afterwards, if that makes sense).

One potential hacky workaround would be to perform said double-parsing within a Run Script operation. Here's an example:

https://user-images.githubusercontent.com/42867097/194992819-b76538cd-0acb-4aff-8458-be1aeb7385fc.mp4

Note that `applyOptionsData` is the same function used by the API to parse mustache syntax variables (double brackets), which is why we're using it. However do note that **this is by no means an official endorsement of this solution**, and it may not work if this function was ever modified substantially in the future. Alternatively, you may use the micromustache package directly with additional logic to avoid such potential issue.

That said, in the case of using [Email Templates extensions](https://docs.directus.io/extensions/email-templates.html) (not via a collection), this PR #15829 will add support for it 👍
